spring-integration/spring-boot-autoconfigure/src/main/java/ai/timefold/solver/spring/boot/autoconfigure/TimefoldAutoConfiguration.java

```diff
@@ -64,62 +70,422 @@
     }
 
     @Override
     public void setBeanClassLoader(ClassLoader beanClassLoader) {
         this.beanClassLoader = beanClassLoader;
     }
 
-    @Bean
-    public TimefoldSolverBannerBean getBanner() {
-        return new TimefoldSolverBannerBean();
+    @Override
+    public void setApplicationContext(ApplicationContext context) throws BeansException {
+        this.context = context;
     }
 
-    @Bean
-    @ConditionalOnMissingBean
-    public SolverConfig solverConfig() {
-        String solverConfigXml = timefoldProperties.getSolverConfigXml();
+    @Override
+    public void setEnvironment(Environment environment) {
+        // postProcessBeanFactory runs before creating any bean, but we need TimefoldProperties. Therefore, we use the
+        // Environment to load the properties
+        BindResult<TimefoldProperties> result = Binder.get(environment).bind("timefold", TimefoldProperties.class);
+        this.timefoldProperties = result.orElseGet(TimefoldProperties::new);
+    }
+
+    @Override
+    public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory) throws BeansException {
+        IncludeAbstractClassesEntityScanner entityScanner = new IncludeAbstractClassesEntityScanner(this.context);
+        if (!entityScanner.hasSolutionOrEntityClasses()) {
+            LOG.warn(
+                    """
+                            Skipping Timefold loading because there are no @%s  or @%s annotated classes.
+                            Maybe your annotated classes are not in a subpackage of your @%s annotated class's package.
+                            Maybe move your planning solution and entity classes to your application class's (sub)package (or use @%s)."""
+                            .formatted(PlanningSolution.class.getSimpleName(), PlanningEntity.class.getSimpleName(),
+                                    SpringBootApplication.class.getSimpleName(), EntityScan.class.getSimpleName()));
+            beanFactory.registerSingleton(DEFAULT_SOLVER_CONFIG_NAME, new SolverConfig(beanClassLoader));
+            return;
+        }
+        Map<String, SolverConfig> solverConfigMap = new HashMap<>();
+        // Step 1 - create all SolverConfig
+        // If the config map is empty, we build the config using the default solver name
+        if (timefoldProperties.getSolver() == null || timefoldProperties.getSolver().isEmpty()) {
+            solverConfigMap.put(TimefoldProperties.DEFAULT_SOLVER_NAME,
+                    createSolverConfig(timefoldProperties, TimefoldProperties.DEFAULT_SOLVER_NAME));
+        } else {
+            timefoldProperties.getSolver().keySet()
+                    .forEach(solverName -> solverConfigMap.put(solverName, createSolverConfig(timefoldProperties, solverName)));
+        }
+        // Step 2 - validate all SolverConfig definitions
+        // TODO - Should we assert planning members belongs to a PlanningEntity Quarkus#assertNoMemberAnnotationWithoutClassAnnotation?
```

Comment:
I suggest we should have the same behavior in Spring Boot and Quarkus, where possible.
What would be the reasons _not_ to do this here?

Reply:
My only concern is whether using reflection can cause issues while generating the native image. Let me replicate the behavior of Quarkus and evaluate it later.